domains/auth/Auth.API/API/Repositories/UserRepository.cs

```diff
@@ -31,4 +39,12 @@ public async Task RemoveUserAsync(User user)
     }
 
     public async Task<IEnumerable<User>> GetUsersByTinAsync(string tin) => await dataContext.Users.Where(x => x.Company != null && x.Company.Tin == tin).Include(u => u.UserRoles).ToListAsync();
+    public async Task<User> UpdateTermsAccepted(User user, DecodableUserDescriptor descriptor)
+    {
+        dataContext.Users.Update(user);
+        await publishEndpoint.Publish(new OrgAcceptedTerms((Guid)user.Id!, descriptor.Organization?.Tin, descriptor.Id));
+
+        await dataContext.SaveChangesAsync();
+        return user;
```

Comment:
Not pretty to have the repo publish events. Can you expose SaveChangesAsync isntead? 
On a sidenote we need UnitOfWork ideally for this